R/module_plots.R

```diff
@@ -7,21 +7,23 @@ plotsUI = function(id) {
   ns = shiny::NS(id)
   # Plots tab panel
   shiny::tabPanel("Plots",
-                  # drop down menu to select plot
-                  shiny::br(),
-                  shiny::uiOutput(ns("choose_plot")),
-
-                  # display plot
-                  display_panel(shiny::uiOutput(ns("display_plot"))),
 
-                  # download button to download current plot
                   shiny::br(),
                   shiny::fluidRow(
-                    shiny::column(12,
+                    # drop down menu to select plot
+                    shiny::column(3,
                                   align = "center",
-                                  shiny::uiOutput(ns("download_plot_button")) #nolint
+                                  shiny::uiOutput(ns("choose_plot")),
+                                  shiny::br(),
+                                  shiny::uiOutput(ns("download_plot_button"))
+                    ),
+                    # display plot
+                    shiny::column(9,
+                                  align = "center",
+                                  display_panel(shiny::uiOutput(ns("display_plot")))
                     )
                   )
+
   )
```

Comment:
You have three (almost) identical copies of this code now. Could you extract a function that creates the UI, and accepts the table title, ID for the choice dropdown, ID for the download button, and the display panel as arguments.
```r
downloader_tab_panel <- function(title, chooser_id, download_button_id, panel) {
    shiny::tabPanel(
       title,
                  shiny::br(),
                  shiny::fluidRow(
                    # drop down menu to select dataset
                    shiny::column(3,
                                  align = "center",
                                  shiny::uiOutput(chooser_id),
                                  shiny::br(),
                                  shiny::uiOutput(download_button_id)
                    ),
                    # display data
                    shiny::column(9, align = "center", panel)
                  )
}
```

Reply:
I've added this function, and updated the modules UI. Thanks - the module code looks much neater now :)